homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -319,6 +344,96 @@ def _async_update_volume(self, data: VolumeState) -> None:
 
         self.async_write_ha_state()
 
+    @callback
```

Comment:
Please remove the callback decorator. This is a coroutine function.

Reply:
OK, so simply removing the decorater is the correct way to handle async callbacks. I have the same issue in https://github.com/home-assistant/core/pull/121209 but haven't gotten an answer.

Reply:
To be clear, this is not an async callback. It's a coroutine function. That's why we shouldn't decorate it with `@callback`.

Reply:
https://developers.home-assistant.io/docs/asyncio_categorizing_functions

Reply:
Got it, thanks. I'll make the necessary changes.

Reply:
I've opened a PR https://github.com/home-assistant/core/pull/126420